src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/FlagVisitsTask.kt

```diff
@@ -87,7 +87,7 @@ class FlagVisitsTask(
       reason = notifications.joinToString(", ") { it.description }
     } else {
       try {
-        sessions = sessionService.getVisitSessions(prisonCode = visit.prisonCode, prisonerId = visit.prisonerId, minOverride = noticeDays, maxOverride = noticeDays)
+        sessions = sessionService.getVisitSessions(prisonCode = visit.prisonCode, prisonerId = visit.prisonerId, minOverride = noticeDays, maxOverride = noticeDays).filter { it.sessionTemplateReference == visit.sessionTemplateReference }
```

Comment:
fix bug where there are multiple sessions on a day and the other session is available to the prisoner after a cell move.